### PR TITLE
ENT-152 Inventory memory on Windows and HPUX

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @craigcomstock can review this?

**Comment by nickanderson** [comment]
@craigcomstock if you can double check/test this it would be great

File: inventory/any.cf [hunk 1]
```diff
@@ -141,22 +141,43 @@ bundle agent cfe_autorun_inventory_memory
...
+@if minimum_version(3.11)
+      # The `with` attribute is necessary for this to work in a single promise.
+    enterprise.windows::
+      "total"
+        meta => { "inventory", "attribute_name=Memory size (MB)" },
+        string => format( "%d", eval("$(with)/1024", "math", "infix" )),
+        if => not( isvariable( "total" ) ),
+        with => nth( string_split( execresult("wmic OS get TotalVisibleMemorySize /format:list", useshell ),
+                                   "=", 2), 1);
+      "totalPhysical"
```
Comment (craigcomstock) [inline inventory/any.cf]: A comment on the first one would be nice. Like:

```
# wmic returns "TotalVisibleMemorySize=10760224" so split on = and take the second item (0-based with nth())
```

File: inventory/any.cf [hunk 2]
```diff
@@ -141,22 +141,43 @@ bundle agent cfe_autorun_inventory_memory
...
-                               $(mon.value_mem_total)),
-      meta => { "inventory", "attribute_name=Memory size (MB)" };
-
-    enterprise::
-      "free" string => ifelse("windows", "0",
-                              $(mon.value_mem_free)),
-      meta => { "report" };
-
+@if minimum_version(3.11)
+      # The `with` attribute is necessary for this to work in a single promise.
```
Comment (craigcomstock) [inline inventory/any.cf]: Shouldn't there be an @else for this minimum_version guard? What happens in our 3.10 LTS for this? Shouldn't it return 0 for total and free as before?

**Comment by craigcomstock** [comment]
> @craigcomstock if you can double check/test this it would be great

Will do!

File: inventory/any.cf (hunk 1 above)
Comment (nickanderson) [reply inventory/any.cf]: ACK

File: inventory/any.cf (hunk 2 above)
Comment (nickanderson) [reply inventory/any.cf]: AFAIK we do not have an @else . The important thing is that the parser will complain on any version older than 3.11 if it sees the with the attribute in use.

reporting 0 is lame.

Comment (craigcomstock) [reply inventory/any.cf]: Ok. No @else. @nickanderson I tested with 3.10 and 3.12 agents and this policy seems fine. I think the difference is not too important.

**Pull request merged**